drivers/volume/portworx/portworx.go

```diff
@@ -2621,67 +2621,69 @@ func (p *portworx) UpdateMigratedPersistentVolumeSpec(
 	if len(pv.Spec.StorageClassName) != 0 {
 		sc, err := storage.Instance().GetStorageClass(pv.Spec.StorageClassName)
 		if err != nil {
-			return nil, fmt.Errorf("failed in getting the storage class [%v]: %v", pv.Spec.StorageClassName, err)
+			logrus.Warnf("failed in getting the storage class [%v]: %v", pv.Spec.StorageClassName, err)
 		}
-		if isCsiProvisioner(sc.Provisioner) {
-			// add csi section in the pv spec
-			if pv.Spec.CSI == nil {
-				pv.Spec.CSI = &v1.CSIPersistentVolumeSource{}
-			}
-			// get the destinationNamespace
-			var dstNamespace string
-			var exists bool
-			if dstNamespace, exists = namespaceMapping[vInfo.SourceNamespace]; !exists {
-				dstNamespace = vInfo.SourceNamespace
-			}
-			// Update the controller expand secret
-			if val, ok := sc.Parameters[controllerExpandSecretName]; ok {
-				if pv.Spec.CSI.ControllerExpandSecretRef == nil {
-					pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+		if sc != nil {
+			if isCsiProvisioner(sc.Provisioner) {
+				// add csi section in the pv spec
+				if pv.Spec.CSI == nil {
+					pv.Spec.CSI = &v1.CSIPersistentVolumeSource{}
 				}
-				if val == templatizedName {
-					pv.Spec.CSI.ControllerExpandSecretRef.Name = vInfo.PersistentVolumeClaim
-				} else {
-					pv.Spec.CSI.ControllerExpandSecretRef.Name = val
+				// get the destinationNamespace
+				var dstNamespace string
+				var exists bool
+				if dstNamespace, exists = namespaceMapping[vInfo.SourceNamespace]; !exists {
+					dstNamespace = vInfo.SourceNamespace
 				}
-			}
-			if val, ok := sc.Parameters[controllerExpandSecretNamespace]; ok {
-				if pv.Spec.CSI.ControllerExpandSecretRef == nil {
-					pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+				// Update the controller expand secret
+				if val, ok := sc.Parameters[controllerExpandSecretName]; ok {
+					if pv.Spec.CSI.ControllerExpandSecretRef == nil {
+						pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+					}
+					if val == templatizedName {
+						pv.Spec.CSI.ControllerExpandSecretRef.Name = vInfo.PersistentVolumeClaim
+					} else {
+						pv.Spec.CSI.ControllerExpandSecretRef.Name = val
+					}
 				}
-				if val == templatizedNamespace {
-					pv.Spec.CSI.ControllerExpandSecretRef.Namespace = dstNamespace
-				} else {
-					pv.Spec.CSI.ControllerExpandSecretRef.Namespace = val
+				if val, ok := sc.Parameters[controllerExpandSecretNamespace]; ok {
+					if pv.Spec.CSI.ControllerExpandSecretRef == nil {
+						pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+					}
+					if val == templatizedNamespace {
```

Comment:
Here asumption is `templatizedNamespace="${pvc.name}"` but if a user chooses something else as this is user choice, one suggestion is to have a Regex here to check for anything starting with `$` and use the val as is. 
When i was searching for this on csi website, i found an example https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html#per-volume-secrets where it is different.

Reply:
Good point. As per our document, we are suggesting this. 
https://docs.portworx.com/operations/operate-kubernetes/storage-operations/csi/security/
To support $(pvc.Name)-xxxxxxx, we need to change in other places as well. We can take it as separate fix.